### DISCUSSION TITLE
Velocity timescale separation cannot divide this interior viscosity timescale separation

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#60] [General discussion]
Hi,

This is my first time running a simulation in V 2.6, when I start running I get this error:

`(HemoCell) Error, Velocity timescale separation cannot divide this interior viscosity timescale separation, exiting ...`

Can anybody help me with it?

**Comment by gzavo** [comment]
Hi @mojtabaAmiraslanpour ,

This should not be the effect of the version change. This is the triggered error message:
https://github.com/UvaCsl/HemoCell/blob/ccccc63aacc211847e79f4478cf701e78663487b/core/hemoCell.cpp#L613-L619
The error means you have internal viscosity enabled, and the steps where you update the internal cells don't coincide with the steps when the cell model is updated (which is a problem).
The time scales are updated through the functions from (and the subsequent functions below this in the same file):
https://github.com/UvaCsl/HemoCell/blob/ccccc63aacc211847e79f4478cf701e78663487b/core/hemoCell.cpp#L384

In most examples these are set from the config.xml.

**Comment by mojtabaAmiraslanpour** [comment]
Thanks @gzavo 
Well, this is strange because I think I have everything right in the config.xml but still, the two numbers are not dividable. This is how my config file is set:

```xml
<ibm>
    <stepMaterialEvery> 5 </stepMaterialEvery> <!-- Update particle material model after this many fluid time steps. -->
    <stepParticleEvery> 5 </stepParticleEvery> <!-- Update particles position after this many fluid time steps. -->
</ibm>

<domain>
	...
</domain>

<sim>
    <tmax> 20000000 </tmax> <!-- total number of iterations -->
    <tmeas>  10000 </tmeas> <!-- interval after which data is written -->
    <tlog>  1000 </tlog> <!-- interval after which data is logged on the screen -->
    <tcsv>  10000 </tcsv> <!-- interval after which simple mean cell values are written to a CSV file -->
    <tcheckpoint> 100000 </tcheckpoint>
    <interiorViscosity>5</interiorViscosity>
    <interiorViscosityEntireGrid>1000</interiorViscosityEntireGrid>
</sim>
```

**Comment by mojtabaAmiraslanpour** [comment]
I found the solution. So in order to add interior viscosity one needs to add the following line into the main code to set _Interior Viscosity Timescale Separation_:

`  hemocell.setInteriorViscosityTimeScaleSeperation((*cfg)["sim"]["interiorViscosity"].read<int>(),(*cfg)["sim"]["interiorViscosityEntireGrid"].read<int>());`

This is better to be mentioned here in the user guide for less future confusion:
https://www.hemocell.eu/user_guide/Case.html#compiling-examples-with-special-features

Have a nice day!
